lib/batch.js

```diff
@@ -0,0 +1,49 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const logger = require('./logging')('batch');
+const P = require('./promise');
+
+function inject(server, options) {
+  return new P(function(resolve) {
+    server.inject(options, resolve);
+  });
+}
+
+function batch(request, map) {
+  var result = {};
+  Object.keys(map).forEach(function(prop) {
+    var url = map[prop];
+    result[prop] = inject(request.server, {
+      method: 'get',
+      url: url,
+      headers: request.headers,
+      credentials: request.auth.credentials
```

Comment:
Does passing this through here mean that we don't have to re-check the credentials on each sub-request?